vms/platformvm/validators/manager.go

```diff
@@ -413,5 +415,42 @@ func (m *manager) OnAcceptedBlockID(blkID ids.ID) {
 }
 
 func (m *manager) GetCurrentValidatorSet(ctx context.Context, subnetID ids.ID) (map[ids.ID]*validators.GetCurrentValidatorOutput, uint64, error) {
-	return m.state.GetCurrentValidatorSet(ctx, subnetID)
+	result := make(map[ids.ID]*validators.GetCurrentValidatorOutput)
+	baseStakers, l1Validators, height, err := m.state.GetCurrentValidators(ctx, subnetID)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	for _, validator := range baseStakers {
+		if err := ctx.Err(); err != nil {
+			return nil, 0, err
+		}
```

Comment:
this check is unneeded - each iteration takes near zero time. You could have a context error check at the end of the function although that would be quite strange still - I would rather not check the context at all past `GetCurrentValidators`

```suggestion
```